lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -90,10 +90,16 @@ def purchase(money, payment, options = {})
 
       def capture(money, authorization, options = {})
         post = {}
+
         add_amount(post, money, options)
         add_application_fee(post, options)
 
-        commit(:post, "charges/#{CGI.escape(authorization)}/capture", post, options)
+        if emv_tc_response = options.delete(:icc_data)
+          post[:card] = {emv_approval_data: emv_tc_response}
+          commit(:post, "charges/#{CGI.escape(authorization)}", post, options)
```

Comment:
This line is duplicated with the `else` below - maybe add `:card` to `post` and then have the `commit` outside?